src/map.cpp

```diff
@@ -955,6 +955,12 @@ void map::unboard_vehicle( const tripoint &p, bool dead_passenger )
     vp->vehicle().invalidate_mass();
 }
 
+struct rider_data {
+    int part = -1;
+    player *psg = nullptr;
+    monster *pet = nullptr;
```

Comment:
Both `player` (or, more generally, `Character`) and `monster` are derived from `Creature`. Why don't use something like `Creature *psg` instead of these two?

Reply:
Quoting myself in the PR message: "It's a little annoying to have two branches everywhere in displace_vehicle(), but as I recall Creature->pos() called from a monster or player cast as a Creature does weird stuff, so I went with verbose and safe over terse and potentially buggy."

Reply:
Argh! Apologies.

*I should always read the description first, I should always read the description first.

Reply:
> Creature->pos() called from a monster or player cast as a Creature does weird stuff, so I went with verbose and safe over terse and potentially buggy

As a side note, this was due to another bug (the monster had been deleted and its Creature vftable was still here). In normal conditions polymorphism still applies and you can safely call Creature->pos() after an upcast.

Reply:
However, the question withstands. I took a good look at them and I see that `pos()` is a pure virtual function in `Creature`. Both `player` and `monster` override it, so the polymorphic invocation should be perfectly safe.

Reply:
okay, I'll fix it - suits me fine to not have to branch twice everywhere.